File: src/backend/AccelerationRecord.java
package backend;

import backend.motion.Acceleration;
import backend.motion.Motion;

public class AccelerationRecord {

    public Motion motion;
    public Acceleration accelerationToAdd;

    public AccelerationRecord(Motion motion, Acceleration accelerationToAdd) {
        this.motion = motion;
        this.accelerationToAdd = accelerationToAdd;
    }

}

File: src/main/kotlin/engine/AccelerationRecord.kt
package engine

import engine.motion.Acceleration
import engine.motion.Force
import engine.motion.Motion

class AccelerationRecord(var motion: Motion, var acceleration: Acceleration) {

    constructor(freeBody: FreeBody, force: Force)
            : this(freeBody.motion, Acceleration(force.x / freeBody.mass, force.y / freeBody.mass))

}
